src/nodes/peerdfs.cc

```diff
@@ -44,27 +31,16 @@ PeerDFS::PeerDFS() : Node() {
 
 PeerDFS::~PeerDFS() { }
 // }}}
-// establish {{{
-bool PeerDFS::establish() {
- logger->info ("Running Eclipse id=%d", id);
- network->establish();
-
- while (not connected) sleep(1);
- return true;
-}
-// }}}
 // insert {{{
 void PeerDFS::insert(uint32_t hash_key, std::string name, std::string v) {
   int which_node = boundaries->get_index(hash_key);
 
   if (which_node == id) {
-    logger->info ("[DFS] Saving locally KEY: %s", name.c_str());
-    string file_path = disk_path + string("/") + name;
-    ofstream file (file_path);
-    file << v;
-    file.close();
+    INFO("[DFS] Saving locally KEY: %s", name.c_str());
+    local_io.write(name, v);
+
   } else {
-    logger->info ("[DFS] Forwaring KEY: %s -> %d", name.c_str(), which_node);
+    INFO("[DFS] Forwaring KEY: %s -> %d", name.c_str(), which_node);
     KeyValue kv (hash_key, name, v);
     network->send(which_node, &kv);
```

Comment:
That sounds like RPC, it would be a good idea, so remoteIO and localIO uses the same interface